packages/base/code-ref.gts

```diff
@@ -35,6 +46,84 @@ class BaseView extends Component<typeof CodeRefField> {
   </template>
 }
 
+class EditView extends Component<typeof CodeRefField> {
+  @consume(CardURLContextName) declare cardURL: string | undefined;
+  @tracked validationState: 'initial' | 'valid' | 'invalid' = 'initial';
+  @tracked private maybeCodeRef: string | undefined = maybeSerializeCodeRef(
+    this.args.model ?? undefined,
+  );
+
+  <template>
+    <BoxelInput
+      data-test-hasValidated={{this.setIfValid.isIdle}}
+      @value={{this.maybeCodeRef}}
+      @state={{this.validationState}}
+      @onInput={{this.onInput}}
+      @disabled={{not @canEdit}}
+    />
+  </template>
+
+  constructor(owner: Owner, args: any) {
+    super(owner, args);
+    if (this.maybeCodeRef != null) {
+      this.setIfValid.perform(this.maybeCodeRef, { checkOnly: true });
+    }
+  }
+
+  private onInput = (inputVal: string) => {
+    this.maybeCodeRef = inputVal;
+    this.setIfValid.perform(this.maybeCodeRef);
+  };
+
+  private setIfValid = restartableTask(
+    async (maybeCodeRef: string, opts?: { checkOnly?: true }) => {
+      this.validationState = 'initial';
+      if (maybeCodeRef.length === 0) {
+        if (!opts?.checkOnly) {
+          this.args.set(undefined);
+        }
+        return;
+      }
+
+      let parts = maybeCodeRef.split('/');
+      if (parts.length < 2) {
+        this.validationState = 'invalid';
+        return;
+      }
+
+      let name = parts.pop()!;
+      let module = parts.join('/');
+      try {
+        if (moduleIsUrlLike(module)) {
+          await loadCard(
```

Comment:
Is this OK to use even if the module you are loading may not be a card?

Reply:
if the code ref is a `ResolvedCodeRef` type then yes--it can be used to load any export. for any other types of code refs then no. However, we are inherently treating the serialized code refs as `ResolvedCodeRef` types.

Reply:
Hmm, looking at the source of `loadCard`, it looks like it throws if the imported things is not a BaseDef via `isBaseDef` check.

Reply:
ah, you're right. let me use the same validation that we use for the non-URL like code refs